packages/backend-dynamic-feature-service/src/schemas/schemas.ts

```diff
@@ -152,8 +157,16 @@ export const dynamicPluginsSchemasServiceFactory = Object.assign(
 async function gatherDynamicPluginsSchemas(
   packages: ScannedPluginPackage[],
   logger: LoggerService,
-  schemaLocator: (pluginPackage: ScannedPluginPackage) => string = () =>
-    path.join('dist', 'configSchema.json'),
+  schemaLocator: (
+    pluginPackage: ScannedPluginPackage,
+  ) => string = pluginPackage =>
+    path.join(
+      'dist',
+      PackageRoles.getRoleInfo(pluginPackage.manifest.backstage.role)
+        .platform === 'node'
+        ? 'configSchema.json'
```

Comment:
Is this logic sound? In the base case, package.json points out the schema file (which can have any name). And should the logic be inverted such that "the special case" of the dotfile is checked instead?

Reply:
You're right.
This should simply point to the compiled schema of the module federation remote for frontend plugins, and to the same file name for compiled schemas of backend plugins for the sake of consistency.
This has been modified in commit [f7886f6](https://github.com/backstage/backstage/pull/28076/commits/f7886f62fdaee092c85d068ceecde51e4bfb99cd#diff-69befd61a07507b883f67e6c0dec7618f5c026d3b70958e6c1d5dda1ccc20722)